### PR TITLE
fix: correct set diff annotations for unordered collections

### PR DESCRIPTION
**Pull request by younes-io**
Related to issue https://github.com/tlaplus/vscode-tlaplus/issues/291

Root cause : the diff algorithm compares sets by position/order (like arrays), but sets are unordered collections. When elements appear in different positions, they are treated as different. 

In this PR:
1. Changed SetValue.diff() to do proper unordered comparison: matches elements by content, not position
2. Updated the test that expected bar → baroque to be "modified" (it's actually delete + add for sets)
3. Fixed property tests that had wrong assumptions about duplicate values in sets
4. Added property-based testing with fast-check: 6 properties with 1000 test runs each to catch edge cases + [1 specific use case](https://github.com/younes-io/vscode-tlaplus/blob/bbb8ae835c1acca684e83e2d9e45c8d79b4de3fd/tests/suite/model/check-set-diff-properties.test.ts#L235) that reproduces https://github.com/tlaplus/vscode-tlaplus/issues/291#issue-1718598274

Sets should now behave like this: reordering doesn't create false changes, and adding new elements doesn't affect existing ones.

### CODE REVIEW

**Review by lemmy — APPROVED** [review]
LGTM (AFAICT it relies on TLC normalizing set elements.  For example, the algo would fail on `{1,2,3}` and `1..3`, because the string representation of this set is different. I think this is fine).

File: src/model/check.ts [hunk 1]
```diff
@@ -281,45 +281,64 @@ export class SetValue extends CollectionValue {
...
-                    notfound = false;
+        const deletedItems: Value[] = [];
+
+        // Build arrays to track which items we've matched
+        const thisMatched = new Array(this.items.length).fill(false);
+        const otherMatched = new Array(o.items.length).fill(false);
+
+        const otherItemMap = new Map<string, number[]>();
+        for (let j = 0; j < o.items.length; j++) {
+            const str = o.items[j].str;
```
Comment (lemmy) [inline src/model/check.ts:294]: Why does this code need to handle multiple indices? I thought it was working with sets, which shouldn't have duplicates. Is there something I'm missing, or is this just a precaution?

Comment (lemmy) [reply src/model/check.ts:294]: Thanks for the explanation! Could you please include your explanation as a code comment? After adding that, the PR will be ready to merge.
